debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerStreamingChangeEventSourceMetrics.java

```diff
@@ -0,0 +1,839 @@
+     */
+    public void calculateLagFromSource(Instant changeTime) {
+        if (changeTime != null) {
+            final Instant adjustedTime = changeTime.plusMillis(timeDifference.longValue())
+                    .minusSeconds(databaseZoneOffset.get().getTotalSeconds());
+            lagFromSourceDuration.set(Duration.between(adjustedTime, clock.instant()).abs());
+        }
+    }
+
+    @Override
+    public String toString() {
+        return "LogMinerStreamingChangeEventSourceMetrics{" +
+                "connectorConfig=" + connectorConfig +
+                ", startTime=" + startTime +
+                ", clock=" + clock +
+                ", currentScn=" + currentScn +
+                ", offsetScn=" + offsetScn +
+                ", commitScn=" + commitScn +
+                ", oldestScn=" + oldestScn +
+                ", oldestScnTime=" + oldestScnTime +
+                ", currentLogFileNames=" + currentLogFileNames +
+                ", redoLogStatuses=" + redoLogStatuses +
+                ", databaseZoneOffset=" + databaseZoneOffset +
+                ", batchSize=" + batchSize +
+                ", logSwitchCount=" + logSwitchCount +
+                ", logMinerQueryCount=" + logMinerQueryCount +
+                ", sleepTime=" + sleepTime +
+                ", minimumLogsMined=" + minimumLogsMined +
+                ", maximumLogsMined=" + maximumLogsMined +
+                ", maxBatchProcessingThroughput=" + maxBatchProcessingThroughput +
+                ", timeDifference=" + timeDifference +
+                ", processedRowsCount=" + processedRowsCount +
+                ", activeTransactionCount=" + activeTransactionCount +
+                ", rolledBackTransactionCount=" + rolledBackTransactionCount +
+                ", oversizedTransactionCount=" + oversizedTransactionCount +
+                ", changesCount=" + changesCount +
+                ", scnFreezeCount=" + scnFreezeCount +
+                ", batchProcessingDuration=" + batchProcessingDuration +
+                ", fetchQueryDuration=" + fetchQueryDuration +
+                ", commitDuration=" + commitDuration +
+                ", lagFromSourceDuration=" + lagFromSourceDuration +
+                ", miningSessionStartupDuration=" + miningSessionStartupDuration +
+                ", parseTimeDuration=" + parseTimeDuration +
+                ", resultSetNextDuration=" + resultSetNextDuration +
+                ", userGlobalAreaMemory=" + userGlobalAreaMemory +
+                ", processGlobalAreaMemory=" + processGlobalAreaMemory +
+                ", abandonedTransactionIds=" + abandonedTransactionIds +
+                ", rolledBackTransactionIds=" + rolledBackTransactionIds +
+                "} " + super.toString();
+    }
+
+    /**
+     * Utility class for tracking histogram-based values for a duration-based metric.
+     */
+    @ThreadSafe
+    static class DurationHistogramMetric {
```

Comment:
There is `io.debezium.connector.spanner.metrics.latency` package. Could you please take a look if it makes sense to move those callses into the main repo and re-use them?